doc/rbd/rbd-snapshot.rst

```diff
@@ -206,107 +232,131 @@ performed these steps, you can begin cloning the snapshot.
-   a content management system, an analytics system, etc.) and then snapshot the 
-   extended image, which itself may be updated just like the base image.
-
-#. **Template Pool:** One way to use block device layering is to create a 
-   pool that contains master images that act as templates, and snapshots of those
-   templates. You may then extend read-only privileges to users so that they 
-   may clone the snapshots without the ability to write or execute within the pool.
-
-#. **Image Migration/Recovery:** One way to use block device layering is to migrate
-   or recover data from one pool into another pool.
+ID, the image ID, and the snapshot ID. The inclusion of the pool ID means that
+you may clone snapshots from one pool to images in another pool.
+
+#. **Image Template:** A common use case for block device layering is to create
+   a master image and a snapshot that serves as a template for clones. For
+   example: a user may create an image for a Linux distribution (for example,
+   Ubuntu 12.04) and create a snapshot for it. The user may occasionally update
+   the image and create a new snapshot (by using such commands as ``sudo
+   apt-get update``, ``sudo apt-get upgrade``, or ``sudo apt-get dist-upgrade``
+   followed by ``rbd snap create``). As the image matures, the user can clone
+   any one of the snapshots.
+
+#. **Extended Template:** A more advanced use case includes extending a
+   template image that provides more information than a base image. For
+   example, a user may clone an image (for example, a VM template) and install
+   other software (for example, a database, a content management system, an
+   analytics system) and then snapshot the extended image, which may itself be
+   updated just like the base image.
+
+#. **Template Pool:** One way to use block device layering is to create a pool
+   that contains both (1) master images that act as templates and (2) snapshots
+   of those templates. You may then extend read-only privileges to users so
+   that they may clone the snapshots without the ability to write or execute
+   within the pool.
+
+#. **Image Migration/Recovery:** One way to use block device layering is to
+   migrate or recover data from one pool into another pool.
 
 Protecting a Snapshot
 ---------------------
 
-Clones access the parent snapshots. All clones would break if a user inadvertently 
-deleted the parent snapshot. To prevent data loss, you **MUST** protect the
-snapshot before you can clone it. ::
+Clones access the parent snapshots. All clones would break if a user
+inadvertently deleted the parent snapshot. To prevent data loss, you **MUST**
+protect the snapshot before you can clone it:
+
+.. prompt:: bash $
 
-	rbd snap protect {pool-name}/{image-name}@{snapshot-name}
+   rbd snap protect {pool-name}/{image-name}@{snapshot-name}
 
-For example::
+For example:
 
-	rbd snap protect rbd/my-image@my-snapshot
+.. prompt:: bash $
+
+   rbd snap protect rbd/my-image@my-snapshot
```

Comment:
Above, the example image and snapshot names were "foo" and "snapname" respectively.  Let's pick one and be consistent.

Reply:
Done. "foo@snapname" it shall be, throughout the document.

%s/my-image@my-snapshot/foo@snapname/g

Reply:
If you chose to settle on "foo", I'd also change the cloned image name in cloning examples to "bar".